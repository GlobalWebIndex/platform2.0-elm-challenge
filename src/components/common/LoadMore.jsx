import React from "react";
import PropTypes from "prop-types";
import { useStyles } from "./LoadMore.style";
import noop from "../../util/noop";

function LoadMore({ onClick = noop, isLoadingMore = false }) {
  const classes = useStyles();

  return (
    <div className={classes.loadMoreWrapper}>
      <button
        disabled={isLoadingMore}
        className={classes.loadMoreButton}
        onClick={onClick}
        data-testid="load-more"
      >
        {isLoadingMore ? `Loading...` : `Load More`}
      </button>
    </div>
  );
}

LoadMore.propTypes = {
  onClick: PropTypes.func,
  isLoadingMore: PropTypes.bool,
};

export default LoadMore;
