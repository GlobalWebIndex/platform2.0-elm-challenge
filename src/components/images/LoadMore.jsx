import React from "react"
import { useLoadMoreStyles } from "./LoadMore.style"

function LoadMore({ onClick }) {
  const classes = useLoadMoreStyles()
  const clickHandler = () => onClick()

  return (
    <div className={classes.loadMoreWrapper}>
      <button className={classes.loadMoreButton} onClick={clickHandler}>
        Load More
      </button>
    </div>
  )
}

export default LoadMore