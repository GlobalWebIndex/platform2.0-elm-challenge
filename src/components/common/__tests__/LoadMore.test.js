import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoadMore from "../LoadMore";

describe("<LoadMore/>", () => {
  it("renders", () => {
    const { container } = render(<LoadMore />);

    expect(container).toMatchSnapshot();
  });

  it("renders while loading", () => {
    const { container } = render(<LoadMore isLoadingMore />);

    expect(container).toMatchSnapshot();
  });

  it("loads more data", () => {
    const onClick = jest.fn();

    render(<LoadMore onClick={onClick} />);

    userEvent.click(screen.getByTestId("load-more"));

    expect(onClick).toHaveBeenCalled();
  });
});
