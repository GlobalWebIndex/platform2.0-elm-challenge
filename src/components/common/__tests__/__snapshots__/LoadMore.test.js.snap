// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMore/> renders 1`] = `
<div>
  <div
    class="loadMoreWrapper-0-2-1"
  >
    <button
      class="loadMoreButton-0-2-2"
      data-testid="load-more"
    >
      Load More
    </button>
  </div>
</div>
`;

exports[`<LoadMore/> renders while loading 1`] = `
<div>
  <div
    class="loadMoreWrapper-0-2-1"
  >
    <button
      class="loadMoreButton-0-2-2"
      data-testid="load-more"
      disabled=""
    >
      Loading...
    </button>
  </div>
</div>
`;
